komodo/yaml_file_types.py

```diff
@@ -38,16 +133,521 @@ class ManifestFile(YamlFile):
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_maintainer(
+        package_name: str, package_version: str, package_maintainer: str
+    ) -> bool:
+        if isinstance(package_maintainer, str):
+            return True
+        raise TypeError(
+            f"Package '{package_name}' version {package_version} has invalid"
+            f" maintainer type ({package_maintainer})"
+        )
+
+    @staticmethod
+    def validate_package_maintainer_with_errors(
+        package_name: str,
+        package_version: str,
+        package_maintainer: str,
+    ) -> List[str]:
+        errors = []
+        try:
+            Package.validate_package_maintainer(
+                package_name, package_version, package_maintainer
+            )
+        except TypeError as e:
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_source(
+        package_name: str, package_version: str, package_source: str
+    ) -> bool:
+        if isinstance(package_source, (str, type(None))):
+            return True
+        raise TypeError(
+            f"Package '{package_name}' version {package_version} has invalid source"
+            f" type ({package_source})"
+        )
+
+    @staticmethod
+    def validate_package_source_with_errors(
+        package_name: str,
+        package_version: str,
+        package_source: str,
+    ) -> List[str]:
+        errors = []
+        try:
+            Package.validate_package_source(
+                package_name, package_version, package_source
+            )
+        except TypeError as e:
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_property_type(
+        package_name: str,
+        package_version: str,
+        package_property: str,
+        package_property_value: str,
+    ):
+        if isinstance(package_property, str):
+            if not package_property.islower():
+                raise ValueError(
+                    f"Package '{package_name}' version '{package_version}' property"
+                    f" should be lowercase ({package_property})"
+                )
+        else:
+            raise TypeError(
+                f"Package '{package_name}' version has invalid property type"
+                f" ({package_property})"
+            )
+        if not isinstance(package_property_value, str):
+            raise TypeError(
+                f"Package '{package_name}' version '{package_version}' property"
+                f" '{package_property}' has invalid property value type"
+                f" ({package_property_value})"
+            )
+
+
+def handle_validation_errors(errors: List[str], message: str):
+    if errors:
+        raise SystemExit("\n".join(errors + [message]))
+
+
+def load_package_status_file(package_status_string: str):
+    package_status = PackageStatusFile().from_yaml_string(package_status_string)
```

Comment:
You can merge these two lines and avoid the declaration and naming of the temporary variable `package_status`

Reply:
Yes, I agree!